bsp/src/mill/bsp/Utils.scala

```diff
@@ -56,24 +56,22 @@ object Utils {
   // Detect and return the test classes contained in the given TestModule
   def getTestClasses(module: TestModule, evaluator: Evaluator)(implicit ctx: Ctx.Home): Seq[String] = {
     val runClasspath = getTaskResult(evaluator, module.runClasspath)
-    val frameworks = getTaskResult(evaluator, module.testFrameworks)
+    val framework = getTaskResult(evaluator, module.testFramework)
     val compilationResult = getTaskResult(evaluator, module.compile)
 
-    (runClasspath, frameworks, compilationResult) match {
-      case (Result.Success(classpath), Result.Success(testFrameworks), Result.Success(compResult)) =>
+    (runClasspath, framework, compilationResult) match {
+      case (Result.Success(classpath), Result.Success(testFramework), Result.Success(compResult)) =>
         val classFingerprint = Jvm.inprocess(
           classpath.asInstanceOf[Seq[PathRef]].map(_.path),
           classLoaderOverrideSbtTesting = true,
           isolated = true,
           closeContextClassLoaderWhenDone = false,
           cl => {
-            val fs = TestRunner.frameworks(testFrameworks.asInstanceOf[Seq[String]])(cl)
-            fs.flatMap(framework =>
-              discoverTests(cl, framework, Agg(compResult.asInstanceOf[CompilationResult].classes.path))
-            )
+            val framework = TestRunner.framework(testFramework.asInstanceOf[String])(cl)
+            discoverTests(cl, framework, Agg(compResult.asInstanceOf[CompilationResult].classes.path))
           }
         )
-        classFingerprint.map(classF => classF._1.getName.stripSuffix("$"))
+        Seq.from(classFingerprint.map(classF => classF._1.getName.stripSuffix("$")))
```

Comment:
Just curiosity: why not `.toSeq`?

Reply:
`.toSeq` gives a deprecation warning, and any fix is more verbose and results in the same call.